### ISSUE TITLE
Out of gas when deploying

### ISSUE THREAD

**Issue by chevdor** [ConsenSys/truffle-solidity-loader#5]
In testrpc@3.0.0, some changes have been made and the gas limit has been lowered.
Using truffle, I could overcome the problem by providing to the truffle.js file:
`rpc.gas:  480000`

This change allows truffle to migrate properly but webpack fails with an "out of gas" error.

I tried to pass the option to the loader as:
`loader: 'truffle-solidity?rpc.gas=480000'`
which seems to have no effect.

As I understand, the loader is supposed to read the truffle config, it does not seem to read rpc.gas

**Comment by chevdor** [comment]
@johnmcdowall are you able to use the loader with testrpc 3.x ?

**Comment by jmdfm** [comment]
Truffle 3.0.0 is not released as of yet, so this module is not developed against it.

**Comment by chevdor** [comment]
Let me comment on an actual solution for this as I spent way too much time looking for it.

First of all, what confused me was that truffle could run the migration without gas issue but webpack could not (using testrpc 3.x which introduces new tx gas limits)

I played with the truffle config and saw that `rpc.gas` seems to be used by truffle but not by webpack/truffle-solidity-loader. I did not see any effect for `networks.xyz.gas`

In order for both truffle and the truffle-solidity-loader to migrate successfully, the solution is to provide the gas in the migration scripts. That can be done per contract:

```
module.exports = function(deployer) {
  deployer.deploy(Migrations, {gas: 155000});
};
```

NOTE: This is a tight limit. With the current contract, 154k will fail. If the contract code is modified, 155k may become too little.

This is actually much nicer like this than to provide a gas limit at the network level (such as done in truffle.js) since a given contract may need much more gas than another.

**Comment by jmdfm** [comment]
@chevdor Thanks for the update - looking at this right now

**Comment by chevdor** [comment]
No problem. Let me know if I can be of any help.

**Comment by jmdfm** [comment]
@chevdor Ok this should be fixed in 0.0.7, where the gas params etc are properly passed along to the migrator. Thanks for reporting this!
